server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -190,6 +190,10 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                 return allocation.decision(YES, NAME, "below primary recovery limit of [%d]", primariesInitialRecoveries);
             }
         } else {
+            if (shardRouting.isSearchOnly()) {
+                // skip any shard type throttling for search replicas and rely only on concurrent incoming node recovery limits.
+                return allocation.decision(YES, NAME, "Do not throttle search replica recovery");
```

Comment:
If we don't control these type of recoveries explicitly then this can result in say primaries of other indices to stay unassigned due to limit hit because these type of recoveries can hog an entire node or multiple nodes. A setting would help us control. What say?